exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -44,7 +44,10 @@ func routeWithDefaults(defaultDSType string) func(
 			}
 		}
 
-		// Receiver-based routing
+		dataset = sanitizeDataStreamDataset(dataset)
+		namespace = sanitizeDataStreamNamespace(namespace)
```

Comment:
Let's do it right at the end when we put things together, on line 65-66, to cover receiver based routing edge cases